internal/indexmanagement/scripts.go

```diff
@@ -131,6 +131,45 @@ def updateWriteIndex(currentIndex, nextIndex, alias):
     print(e)
     sys.stdout = original_stdout
     return False
+
+def getCurrentDiskPercent():
+  original_stdout = sys.stdout
+  try:
+    es_client = getEsClient()
+    response = es_client.cat.allocation(format="JSON", h="disk.percent,node", v=True)
+    diskPercent = []
+    for i in range(len(response)):
+      diskPercent.append(int(response[i]["disk.percent"]))
+    currentDiskPercent = max(diskPercent)
+    return currentDiskPercent
+  except Exception as e:
+    sys.stdout = open('/tmp/response.txt', 'w')
+    print(e)
+    sys.stdout = original_stdout
+    return False
+
+def deleteByPercentage(alias):
+  original_stdout = sys.stdout
+  try:
+    es_client = getEsClient()
+    indices = es_client.cat.indices(index=alias+"*",format="json",h="index,creation.date")
+
+    currentDiskPercent = getCurrentDiskPercent()
+    diskThreshold = os.environ['DISK_THRESHOLD']
+    
+    while (int(currentDiskPercent) >= int(diskThreshold)):
```

Comment:
Why a while loop over here? It might take time for ES to delete the indices and free up the disk usage.
IMO we should just run once and delete a single index. If the disk usage hasn't reached below threshold still then it will be done when the cron job runs next time. WDYT?

Reply:
The way I looked at it is to free up space as soon as possible before more data is indexed between the two rounds of cron jobs, which might be an issue if the jobs don't run as frequently as every 15 mins like in our examples

Reply:
But the thing is does ES free up the disk space as soon as you delete the index? Can you test this case and verify once?
I am telling so because I feel ES takes some time to free up the space. If that happens then because of running while loop we might delete too many indices.

Reply:
You are right, I totally forgot about this. I'll update the code